python/mlx/optimizers/optimizers.py

```diff
@@ -104,7 +124,6 @@ def state(self):
 
     @state.setter
     def state(self, state: dict):
-        self._initialized = True
         self._state = state
```

Comment:
Should we also set the optimizer as not initialized? I think this will be more consistent behavior.

Reply:
You mean explicitly set it to `False` here?

Reply:
Yep. You don't agree? For instance in the case where we don't recreate the optimizer from scratch but load a saved state...

Reply:
I like that. I think it's better to be explicit.